astropy/visualization/wcsaxes/wcsapi.py

```diff
@@ -101,17 +105,24 @@ def custom_ucd_coord_meta_mapping(mapping, *, overwrite=False):
     <BLANKLINE>
     >
     """
+    added_keys: list[str] = []
+    overwritten: dict[str, Any] = {}
     for k, v in mapping.items():
         k = k.removeprefix("custom:")
-        if not overwrite and k in CUSTOM_UCD_COORD_META_MAPPING:
-            raise ValueError(f"UCD metadata mapping {k} already exists.")
+        if k in CUSTOM_UCD_COORD_META_MAPPING:
+            if not overwrite:
+                raise ValueError(f"UCD metadata mapping {k} already exists.")
+            overwritten[k] = CUSTOM_UCD_COORD_META_MAPPING[k]
+        else:
+            added_keys.append(k)
         CUSTOM_UCD_COORD_META_MAPPING[k] = v
 
-    yield
-
-    for k in mapping.keys():
-        k = k.removeprefix("custom:")
-        del CUSTOM_UCD_COORD_META_MAPPING[k]
+    try:
+        yield
+    finally:
+        for k in added_keys:
```

Comment:
Why delete the keys before updating?

Reply:
order between these operations doesn't matter, but they both need to be performed.

Reply:
Ah, should have seen that!